rust/nasl-builtin-cryptographic/tests/aes_cbc.rs

```diff
@@ -5,120 +5,71 @@
 
     #[test]
     fn aes128_cbc_crypt() {
-        let code = r#"
-        key = hexstr_to_data("00000000000000000000000000000000");
-        data = hexstr_to_data("80000000000000000000000000000000");
-        iv = hexstr_to_data("00000000000000000000000000000000");
-        crypt = aes128_cbc_encrypt(key: key, data: data, iv: iv);
-        aes128_cbc_decrypt(key: key, data: crypt, iv: iv);
-        "#;
-        let register = Register::default();
-        let binding = ContextFactory::default();
-        let context = binding.build(Default::default(), Default::default());
-        let mut parser = CodeInterpreter::new(code, register, &context);
-        parser.next();
-        parser.next();
-        parser.next();
-        assert_eq!(
-            parser.next(),
-            Some(Ok(NaslValue::Data(
-                decode_hex("3ad78e726c1ec02b7ebfe92b23d9ec34").unwrap()
-            )))
+        let mut t = TestBuilder::default();
+        t.run(r#"key = hexstr_to_data("00000000000000000000000000000000");"#);
+        t.run(r#"data = hexstr_to_data("80000000000000000000000000000000");"#);
+        t.run(r#"iv = hexstr_to_data("00000000000000000000000000000000");"#);
+        t.ok(
+            r#"crypt = aes128_cbc_encrypt(key: key, data: data, iv: iv);"#,
+            decode_hex("3ad78e726c1ec02b7ebfe92b23d9ec34").unwrap(),
         );
-        assert_eq!(
-            parser.next(),
-            Some(Ok(NaslValue::Data(
-                decode_hex("80000000000000000000000000000000").unwrap()
-            )))
+        t.ok(
+            r#"aes128_cbc_decrypt(key: key, data: crypt, iv: iv);"#,
+            decode_hex("80000000000000000000000000000000").unwrap(),
         );
     }
 
     #[test]
     fn aes192_cbc_crypt() {
-        let code = r#"
-        key = hexstr_to_data("000000000000000000000000000000000000000000000000");
-        data = hexstr_to_data("1b077a6af4b7f98229de786d7516b639");
-        iv = hexstr_to_data("00000000000000000000000000000000");
-        crypt = aes192_cbc_encrypt(key: key, data: data, iv: iv);
-        aes192_cbc_decrypt(key: key, data: crypt, iv: iv);
-        "#;
-        let register = Register::default();
-        let binding = ContextFactory::default();
-        let context = binding.build(Default::default(), Default::default());
-        let mut parser = CodeInterpreter::new(code, register, &context);
-        parser.next();
-        parser.next();
-        parser.next();
-        assert_eq!(
-            parser.next(),
-            Some(Ok(NaslValue::Data(
-                decode_hex("275cfc0413d8ccb70513c3859b1d0f72").unwrap()
-            )))
+        let mut t = TestBuilder::default();
+        t.run(r#"key = hexstr_to_data("000000000000000000000000000000000000000000000000");"#);
+        t.run(r#"data = hexstr_to_data("1b077a6af4b7f98229de786d7516b639");"#);
+        t.run(r#"iv = hexstr_to_data("00000000000000000000000000000000");"#);
+        t.ok(
```

Comment:
Therefore I'm wondering if it would be plausible to remove the Default implementation of `let mut t = TestBuilder::default();` and replace it with a macro additionally adding file!() and line!() into TestBuilder. Although it would still shadow the exact line of the failing test at least someone could immediately figure out where the TestBuilder got created and with the help of the printed nasl code immediately find it.

Reply:
>  Therefore I'm wondering if it would be plausible to remove the Default implementation of let mut t = TestBuilder::default(); and replace it with a macro additionally adding file!() and line!() into TestBuilder. Although it would still shadow the exact line of the failing test at least someone could immediately figure out where the TestBuilder got created and with the help of the printed nasl code immediately find it.

That's a really nice idea. I'll look into it

Reply:
Found a really awesome mechanism in rust to do this even more cleanly:

https://rustc-dev-guide.rust-lang.org/backend/implicit-caller-location.html

Using this, the new error messages point directly to the line of code adding the line of NASL code that fails:

```
Mismatch at nasl-builtin-regex/tests/regex.rs:137:11.
In code "eregmatch(string: string, pattern: "Bar|true", find_all: TRUE);":
Expected: Ok(Array([String("Baz"), String("true")]))
Found:    Ok(Array([String("Bar"), String("true")]))
```